reposerver/repository/repository.go

```diff
@@ -447,8 +448,49 @@ func newEnv(q *apiclient.ManifestRequest, revision string) *v1alpha1.Env {
 	}
 }
 
+func mergeSourceParameters(source *v1alpha1.ApplicationSource, path string) error {
+	appFilePath := filepath.Join(path, ".argocd-app.yaml")
+	info, err := os.Stat(appFilePath)
+	if os.IsNotExist(err) {
+		return nil
+	} else if info != nil && info.IsDir() {
+		return nil
+	} else if err != nil {
+		return err
+	}
+	patch, err := json.Marshal(source)
+	if err != nil {
+		return err
+	}
+
+	data, err := ioutil.ReadFile(appFilePath)
+	if err != nil {
+		return err
+	}
+	data, err = yaml.YAMLToJSON(data)
+	if err != nil {
+		return err
+	}
+	data, err = jsonpatch.MergePatch(data, patch)
+	if err != nil {
+		return err
+	}
+	var merged v1alpha1.ApplicationSource
+	err = json.Unmarshal(data, &merged)
+	if err != nil {
+		return err
+	}
+	*source = merged
+	return nil
+}
+
 // GetAppSourceType returns explicit application source type or examines a directory and determines its application source type
 func GetAppSourceType(source *v1alpha1.ApplicationSource, path string) (v1alpha1.ApplicationSourceType, error) {
+	err := mergeSourceParameters(source, path)
+	if err != nil {
+		return "", err
```

Comment:
I think we should provide some more details when `mergeSourceParameters()` fails and let the user know that the reason of failure was actually a failed merge of `.argocd-app.yaml`, i.e.

`return "", fmt.Errorf("error while merging .argocd-app.yaml: %v", err)`

or something along these lines.